subprojects/gradle/instrumentation-tests/src/integTest/kotlin/com/avito/instrumentation/reservation/client/kubernetes/KubernetesReservationClientIntegrationTest.kt

```diff
@@ -0,0 +1,132 @@
+package com.avito.instrumentation.reservation.client.kubernetes
+
+import com.avito.instrumentation.reservation.client.ReservationClient
+import com.avito.instrumentation.reservation.request.Device.CloudEmulator
+import com.avito.instrumentation.reservation.request.Reservation
+import com.avito.instrumentation.reservation.request.createStubInstance
+import com.avito.test.gradle.getTestProperty
+import com.avito.truth.assertThat
+import com.avito.utils.logging.FakeCILogger
+import com.google.common.truth.Truth.assertThat
+import io.fabric8.kubernetes.client.KubernetesClientException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import java.net.UnknownHostException
+
+internal class KubernetesReservationClientIntegrationTest {
```

Comment:
Should "Clear leaked K8S deployments" know about these fake deployments?

Reply:
yes, in theory this logic about deployments should be shared;
i see "clear leaked deployments" as obsolete thing after introducing shared service, but for now it must be done
thanks